src/Testcontainers.Oracle/OracleBuilder.cs

```diff
@@ -83,6 +108,18 @@ protected override void Validate()
     {
         base.Validate();
 
+        const string message = "The image '{0}' does not support configuring the database. It is only supported on Oracle 18 and onwards.";
+
+        Predicate<OracleConfiguration> databaseConfigurationNotSupported = value =>
+            value.Database != null && value.Image.MatchVersion(v => v.Major < 18);
```

Comment:
Shouldn't the predicate check if images < 18 use the default service (database) name? It should only throw an exception if the service name does not match the default name, right?

Reply:
Technically, yes. But the documentation of the `WithDatabase` method states this:
> The database can only be set for Oracle 18 and onwards.

So an exception is to be expected (even if it would probably work with the default database name).

Reply:
All right! 👍 We don't need to make it more complex than it already is. Until now, I have tried to avoid providing multiple builders, but here it may make sense for the future.